tests/codebuild/run_test.sh

```diff
@@ -112,3 +112,54 @@ function build_fsx_from_s3()
 
    export FSX_ID=$FSX_ID
 }
+
+function delete_all_tests()
+{
+    # Stop jobs so we can do PrivateLink test.
+    kubectl delete hyperparametertuningjob --all
+    kubectl delete trainingjob --all
+    kubectl delete batchtransformjob --all
+    kubectl delete hostingdeployment --all
+    kubectl delete model --all
+}
+
+# Delete a K8s resource and ensure all AWS resources were cleaned up successfully
+# Parameter:
+#    $1: CRD type
+#    $2: Filename of test
+#    $3: (Optional) Timeout
+# e.g. verify_delete trainingjob xgboost-mnist.yaml ".status.sageMakerTrainingJobName"
+function verify_delete()
+{
+   local crd_type="$1"
+   local file_name="$2"
+   local timeout=${3:-"30s"}
+
+   # Apply file and get name
+   local job_name=$(kubectl apply -f $file_name -o json | jq -r ".metadata.name")
+
+   # Wait until job has started
+   timeout "${timeout}" bash -c \
+      'until [ "$(kubectl get "$0" "$1" -o=custom-columns=STATUS:.status | grep -i "$2" | wc -l)" -eq "1" ]; do \
+          sleep 5; \
+       done' "$crd_type" "$job_name" "InProgress"
+   
+   if [ $? -ne 0 ]; then
+     echo "[FAILED] ${crd_type} ${job_name} job did not start within ${timeout}"
+     exit 1
+   fi
+
+   # Check that we can detect the status changing
+   kubectl delete -f "$file_name" &
+   timeout "${timeout}" bash -c \
+      'until [ "$(kubectl get "$0" "$1" -o=custom-columns=STATUS:.status | grep -i "$2" | wc -l)" -eq "1" ]; do \
+          sleep 5; \
+       done' "$crd_type" "$job_name" "Stopping"
+   
+   if [ $? -ne 0 ]; then
+     echo "[FAILED] ${crd_type} ${job_name} job did not start within ${timeout}"
+     exit 1
+   fi
+
+   echo "[PASSED] Verified ${crd_type} ${job_name} deleted successfully"
```

Comment:
So this function verifies that the SageMaker resource (as reported by k8s) is `Stopping`. Why do we not check against AWS proper? I want to make this a solid test that ignores any smart/buggy logic in the operator and confirms that the resources were deleted.

Reply:
I initially started doing this but the messy part was extracting the job name from the spec considering they all have different JSON paths for their name. I wanted to make it a fairly compact test, but that wasn't really possible. My thought was that if we are detecting the state from SageMaker that our controller picking up a "Stopping" state would only happen if it picked it up from the SM resource directly.

Reply:
I see. I mean this test is certainly better than what we had before.

> My thought was that if we are detecting the state from SageMaker that our controller picking up a "Stopping" state would only happen if it picked it up from the SM resource directly

This is true AFAIK, but we sometimes do smart things like reporting a certain state to make the k8s experience nice while doing things in the background. i.e. in HostingDeployment updates / deletes aren't supported when the SM endpoint is "Creating", so we lie and say "Deleting..." even though the actual status is "Creating". I realize you are not testing the HostingDeployment here; my concern is that future us might have to do something similar in one or more of the job controllers. At that point, this test is no longer useful in asserting that SageMaker resources are cleaned up. Does this make sense?